docs/float8.md

```diff
@@ -16,3 +16,16 @@ CONFIG_FILE="./train_configs/llama3_8b.toml" ./run_llama_train.sh --float8.enabl
 For parallelisms, we support float8 all-gather for FSDP (optional) and for TP (by default for `Float8Linear`).
 
 For scaling strategy, we currently support tensor-wise scaling with dynamic scales, and are actively working on tensor-wise scaling with delayed scales. Row-wise scaling is under exploration.
+
+## Why Composing Float8 with `torch.distributed`
+As shown below, for float8 for matmul, `torch._scaled_mm` requires both float8 tensor and scales. Scales are calculated from `max(abs)` of a high precision tensor.
+```
+# float32/bfloat16 matmul, `torch.mm(input, weight)`, does not require scales
+# float8 matmul requires scales to ensure values can
+# fit within the representable range
+torch._scaled_mm(input_fp8, weight_fp8, scale_a=scale_input, scale_b=scale_weight)
+```
+
+Without considering distributed, we cast input and weight into float8 inside forward before calling torch._scaled_mm.
+
+Considering FSDP, we can cast high precision weights (1/N on each rank) into float8, and perform float8 all-gather to save bandwidth. At the beginning of the forward, we already have the unsharded float8 weights. Similarly, we can do float8 all-gather in TP to save bandwidth. The overhead is communicating `max(abs)` across ranks but for llama herd models, we have net wins in training speed.
```

Comment:
I think we can clarify this part.
1. For both FSDP and TP, if we perform float8 all-gather, then we must also perform some communication of scales. As written, only the TP part mentions this.
2. For TP, when would float8 all-gather not be a net win? What part of "llama herd models" matters?

Reply:
> As written, only the TP part mentions this

good point. will mention the communication for both FSDP and TP

> For TP, when would float8 all-gather not be a net win? What part of "llama herd models" matters?

float8 for TP should be strictly better. The amax communication overhead is meant for FSDP. llama herd models is probably less relevant. I can rephrase it to be more accurate, eg, "float8 all gather and amax communication can be a net win over float32/bfloat16 allgather, depending on world size and messge size."